sql/src/main/java/io/crate/analyze/relations/UnionSelect.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.analyze.relations;
+
+import io.crate.analyze.QuerySpec;
+import io.crate.analyze.symbol.Field;
+import io.crate.exceptions.ColumnUnknownException;
+import io.crate.metadata.Path;
+import io.crate.metadata.table.Operation;
+import io.crate.sql.tree.QualifiedName;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+public class UnionSelect implements QueriedRelation {
+
+    private final boolean distinct;
+    private final QuerySpec querySpec = new QuerySpec();
+    private QueriedRelation left;
+    private QueriedRelation right;
+    private QualifiedName name;
+
+    public UnionSelect(QueriedRelation left, QueriedRelation right, boolean distinct) {
+        this.left = left;
+        this.right = right;
+        this.distinct = distinct;
+        this.name = QualifiedName.of("union", left.getQualifiedName().toString(), right.getQualifiedName().toString());
+    }
+
+    public boolean isDistinct() {
+        return distinct;
+    }
+
+    public QueriedRelation left() {
+        return left;
+    }
+
+    public QueriedRelation right() {
+        return right;
+    }
+
+    public void left(QueriedRelation left) {
```

Comment:
why do we need setters here?

Reply:
https://github.com/crate/crate/pull/6515/files#diff-25d85154b2a99f88ee2b45cf3937e416R109

Reply:
Would've liked it if we could've kept this mostly immutable - when we start re-using the analysis for prepared statements it'd be good to ensure that it isn't mutated.

But I guess since the querySpec is also still here we can keep this for now.